### PR TITLE
[Quorum Store] reduce false positives of timeout and low vote count

### PR DESCRIPTION
**Pull request by bchocho**
### Description

Late votes will re-init batch metadata. We assume if a batch hasn't completed and hasn't received a self vote that they are created due to late votes, and don't log their metrics.

### CODE REVIEW

**Review by igor-aptos — APPROVED** [review]
if this is happening, should we increase the timeout, at least for tracking of the metrics?

**Comment by bchocho** [comment]
But we probably only want to know how many votes were received until the timeout, not until some time after the timeout?

File: consensus/src/quorum_store/proof_coordinator.rs [hunk 1]
```diff
@@ -74,6 +76,9 @@ impl IncrementalProofState {
                     .is_none()
                 {
                     self.aggregated_voting_power += voting_power as u128;
+                    if signer == self.info.author() {
+                        self.self_voted = true;
```
Comment (zekun000) [inline consensus/src/quorum_store/proof_coordinator.rs:80]: this is a bit ugly, we can restrict the init to be from self vote only

Comment (bchocho) [reply consensus/src/quorum_store/proof_coordinator.rs:80]: But the concern was that the remote sends us a vote before we do

Comment (zekun000) [reply consensus/src/quorum_store/proof_coordinator.rs:80]: yes, but isn't that very unlikely given it needs to go through a network round trip compared to local channels?

Comment (bchocho) [reply consensus/src/quorum_store/proof_coordinator.rs:80]: This is rare but not extremely unlikely, given that on both local and remote the batch has to be persisted. It was observed in Forge, where the network trip is short, and our other networks do have at least some peers with short round trips.

Comment (zekun000) [reply consensus/src/quorum_store/proof_coordinator.rs:80]: hmm that's very surprising

**Pull request merged**